File: leetcode/121.py
class Solution:
    ddef maxProfit(sef, prices: List[int]) -> int:
        
        # Solution 1 -> Brute-Force Solution

        max_profit = 0
        
        for buy in range(len(prices)-1): # we do not need to buy in a last day
            for sell in range(buy+1, len(prices)):
                profit = prices[sell] - prices[buy]

                if profit > max_profit:
                    max_profit = profit
        return max_profit

File: Java/121.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int res = 0;
        int lowest = prices[0];

        for (int i = 0; i < prices.length; i++){
            if (prices[i] < lowest){
                lowest = prices[i];
            }
            res = Math.max(res, prices[i] - lowest);
        }

        return res;

    }
}